openedx_external_enrollments/tests/external_enrollments/tests_icc_external_enrollment.py

```diff
@@ -66,3 +73,248 @@ def test_str(self):
             self.base.__str__(),
             'ICC',
         )
+
+    @patch('openedx_external_enrollments.external_enrollments.icc_external_enrollment.configuration_helpers')
+    def test_get_random_string(self, configuration_helpers_mock):
+        """
+        Testing _get_random_string method.
+        """
+        USERNAME_SUFFIX_LENGTH = 5
+        configuration_helpers_mock.get_value.return_value = 'setting_value'
+        self.assertEqual(
+            len(self.base._get_random_string(USERNAME_SUFFIX_LENGTH)),  # pylint: disable=protected-access
+            USERNAME_SUFFIX_LENGTH
+        )
+
+        self.assertEqual(
+            self.base._get_random_string(8),  # pylint: disable=protected-access
+            'setting_value'
+        )
+
+    def test_get_icc_user_from_xml_response(self):
+        """
+        Testing get_icc_user_from_xml_response method.
+        """
+        not_found_user_xml_response = '''<?xml version="1.0" encoding="UTF-8" ?>
+        <RESPONSE>
+            <SINGLE>
+                <KEY name="users">
+                    <MULTIPLE></MULTIPLE>
+                </KEY>
+                <KEY name="warnings">
+                    <MULTIPLE></MULTIPLE>
+                </KEY>
+            </SINGLE>
+        </RESPONSE>'''
+        response = Mock()
+        response.content = not_found_user_xml_response
+
+        self.assertEqual(
+            self.base._get_icc_user_from_xml_response(response, 'get_user'),  # pylint: disable=protected-access
+            {}
+        )
+
+        found_user_xml_response = '''<?xml version="1.0" encoding="UTF-8" ?>
```

Comment:
This 'found_user' test must be wrapped in a single test case; We usually test one case at a time. @ivanvgh

Reply:
I was separated into 2 test cases.